src/main/java/org/apache/sysds/runtime/instructions/fed/ReorgFEDInstruction.java

```diff
@@ -50,20 +77,196 @@ public static ReorgFEDInstruction parseInstruction ( String str ) {
 	@Override
 	public void processInstruction(ExecutionContext ec) {
 		MatrixObject mo1 = ec.getMatrixObject(input1);
-		
+		ReorgOperator r_op = (ReorgOperator) _optr;
+
 		if( !mo1.isFederated() )
 			throw new DMLRuntimeException("Federated Reorg: "
 				+ "Federated input expected, but invoked w/ "+mo1.isFederated());
 
-		//execute transpose at federated site
-		FederatedRequest fr1 = FederationUtils.callInstruction(instString, output,
-			new CPOperand[]{input1}, new long[]{mo1.getFedMapping().getID()});
-		mo1.getFedMapping().execute(getTID(), true, fr1);
+		if(instOpcode.equals("r'")) {
+			//execute transpose at federated site
+			FederatedRequest fr1 = FederationUtils.callInstruction(instString,
+				output,
+				new CPOperand[] {input1},
+				new long[] {mo1.getFedMapping().getID()});
+			mo1.getFedMapping().execute(getTID(), true, fr1);
+
+			//drive output federated mapping
+			MatrixObject out = ec.getMatrixObject(output);
+			out.getDataCharacteristics().set(mo1.getNumColumns(), mo1.getNumRows(), (int) mo1.getBlocksize(), mo1.getNnz());
+			out.setFedMapping(mo1.getFedMapping().copyWithNewID(fr1.getID()).transpose());
+		}
+		else if (instOpcode.equals("rdiag")) {
+			AbstractMap.SimpleEntry<FederationMap, Map<FederatedRange, int[]>> result;
```

Comment:
I have never seen the AbstractMap.SimpleEntry being used.
Usually i see Map.Entry<>.
Is there a difference?

Also since the usage of this entry, is to enable  multi returns on functions calls, i would suggest making a small class inside this file, that you use to setup and return the values.